tools/junit-openfeature/src/test/java/dev/openfeature/contrib/tools/junitopenfeature/StringFlagTest.java

```diff
@@ -0,0 +1,174 @@
+package dev.openfeature.contrib.tools.junitopenfeature;
+
+import dev.openfeature.sdk.Client;
+import dev.openfeature.sdk.OpenFeatureAPI;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Execution(ExecutionMode.SAME_THREAD)
+class StringFlagTest {
+
+    private static final String FLAG = "string-flag";
+    private static final String FALLBACK = "fallback";
+    private static final String FLAG_VALUE = "true";
+    private static final String FLAG_VALUE_ALTERNATIVE = "false";
+    private static final String SPECIFIC_DOMAIN = "testSpecific";
+
+
+    @Nested
+    class SimpleConfig {
+
+        @Nested
+        @Flag(name = FLAG, value = FLAG_VALUE, valueType = String.class)
+        @Flag(name = FLAG + "2", value = FLAG_VALUE, valueType = String.class)
+        @Flag(name = FLAG + "3", value = FLAG_VALUE, valueType = String.class)
+        class onClass {
+            @Test
+            void multipleFlagsSimple() {
```

Comment:
What's the difference between this test and the one on line 50 of this file?

More generally, given we're testing multiple flags, would it be reasonable to combine this into a multi-type test to reduce duplication? (e.g. one int, one bool, etc) or do you think that's erroding coverage somewhere?

Reply:
This one tests the setting on the class level vs. line 50 directly on the test method. I wanted to allow the user to be flexible and reduce the boiler code as much as possible. For example, define the most common value on the class level but overwrite it where necessary on the method level.

I know the tests are verbose, but with annotation, I can't work with inheritance. Method calls, even if they're static, are not allowed. Hence, I need to duplicate the code, but I want to ensure that everything works in all kinds of different scenarios. But I agree, it would be good to have another test where we're mixing the types. Maybe my testing is too exhaustive ;)